libass/ass.h

```diff
@@ -324,7 +324,8 @@ void ass_set_extract_fonts(ASS_Library *priv, int extract);
  *   ScaledBorderAndShadow=yes
  *
  * \param priv library handle
- * \param list NULL-terminated list of strings
+ * \param list NULL-terminated list of strings,
+ *             copied and never modified by libass
  */
 void ass_set_style_overrides(ASS_Library *priv, char **list);
```

Comment:
Consider also:

```suggestion
#ifdef __cplusplus
void ass_set_style_overrides(ASS_Library *priv, const char *const *list);
#else
void ass_set_style_overrides(ASS_Library *priv, char **list);
#endif
```

Reply:
Isn't this non-compatible?

Reply:
Only in C—hence the ifdef.

Reply:
To elaborate: C++ allows implicitly constifying any number of outermost layers of a pointer. Hence `const char *const *list`, not `const char **list`. With the latter, a call like `char *x; foo(&x);` could reassign `x = <pointer to some truly immutable memory>;`, which the caller would then treat as writable memory and crash. With the outer layer constified as well, the called function can’t reassign the caller’s pointer, so there’s nothing to worry about. For some reason, C has never implemented this (IMO it totally should, but I assume some kind of inertia prevents it).

Reply:
does this interact in any way with the enclosing `extern "C"` block?

Reply:
No (as far as I know). `extern "C"` affects symbol linkage but not const/volatile conversions.

Reply:
Just to be super sure, I’ve verified that this compiles without any issues using `g++ -Wall -Wextra -pedantic`:

```c++
extern "C" void foo(const char *const *bar);
char *x;
int main() {
    foo(&x);
}
```

Reply:
seems good then; adopted